### PR TITLE
ensure pkg/util/mount compiles & crosses

### LINKED ISSUE #30795: Compilation error on OSX: GetDeviceNameFromMount repeat in pkg/util/mount/mount_unsupported.go
Compilation error in pkg/util/mount/mount_unsupported.go. GetDeviceNameFromMount declared twice. #30724 and #30666 are duplicates and one of them need to be reverted. (build is disabled for linux, that's why both of them got in)

### PR DESCRIPTION
**Pull request by MHBauer**
**What this PR does / why we need it**:
osx doesn't build
**Which issue this PR fixes** _(optional, in `fixes #<issue number>(, #<issue_number>, ...)` format, will close that issue when PR gets merged)_: fixes #30795

**Special notes for your reviewer**:

**Release note**:

``` release-note-none
```
- move compile time check from linux code to generic code
- remove duplication of #30724 and #30666 to resolve #30795

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/30802)

### CODE REVIEW

**Comment by MHBauer** [comment]
@saad-ali & @jfrazelle the previous LGTM'ers.

**Comment by smarterclayton** [comment]
ok to test

**Comment by MHBauer** [comment]
```
# make WHAT=pkg/util/mount GOFLAGS=-v
+++ [0817 19:37:51] Building go targets for linux/amd64:
    pkg/util/mount

λ make WHAT=pkg/util/mount GOFLAGS=-v
+++ [0817 12:38:19] Building go targets for darwin/amd64:
    pkg/util/mount
```

**Comment by jfrazelle** [comment]
@k8s-bot add to whitelist

**Comment by saad-ali** [comment]
LGTM

**Comment by saad-ali** [comment]
This doesn't need to be cherry-picked since only #30724 was cherry picked (https://github.com/kubernetes/kubernetes/pull/30824) not  #30666

**Comment by ixdy** [comment]
P2 cause I want to fix the crossbuild so we can start a `make release` build again.

**Comment by ixdy** [comment]
un-LG. https://github.com/kubernetes/kubernetes/pull/30996 removed the duplicate method.

File: pkg/util/mount/mount_unsupported.go [hunk 1]
```diff
@@ -48,10 +48,6 @@ func (mounter *Mounter) PathIsDevice(pathname string) (bool, error) {
 	return true, nil
 }
 
-func (mounter *Mounter) GetDeviceNameFromMount(mountPath, pluginDir string) (string, error) {
```
Comment (luxas) [inline pkg/util/mount/mount_unsupported.go]: rebase this, it's removed already

**Comment by MHBauer** [comment]
I still think the check needs to move to avoid this happening in the future.

**Comment by saad-ali** [comment]
Yep, need this compile-time check to prevent unsupported from breaking again.

LGTM

**Pull request merged**